File: app/src/main/java/com/adlab/balda/model/GamePlayer.java
package com.adlab.balda.model;

import java.util.ArrayList;
import java.util.List;

public class GamePlayer {

    private int score;
    private List<String> enteredWords;

    public GamePlayer() {
        score = 0;
        enteredWords = new ArrayList<>();
    }

    public void increaseScore(int delta) {
        score += delta;
    }

    public int getScore() {
        return score;
    }

    public void addEnteredWord(String word) {
        enteredWords.add(word);
    }

    public List<String> getEnteredWords() {
        return enteredWords;
    }
}

File: app/src/main/java/com/adlab/balda/model/GamePlayer.kt
package com.adlab.balda.model

import java.util.*

data class GamePlayer(val nickname: String = "Player") {
    var score = 0
        private set

    val enteredWords: MutableList<String> = ArrayList()

    fun addEnteredWord(word: String) {
        enteredWords.add(word)
        score += word.length
    }
}